migration/incremental/032-logical-epoch-to-time-epoch.sql

```diff
@@ -0,0 +1,96 @@
+CREATE TABLE _prom_catalog.global_epoch (
+    current_epoch TIMESTAMPTZ NOT NULL,
+    delete_epoch TIMESTAMPTZ NOT NULL,
+    -- force there to only be a single row
+    is_unique BOOLEAN NOT NULL DEFAULT true CHECK (is_unique = true),
+    UNIQUE (is_unique)
+);
+GRANT SELECT ON TABLE _prom_catalog.global_epoch TO prom_reader;
+GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE _prom_catalog.global_epoch TO prom_writer;
+
+DO $block$
+    DECLARE
+        _is_restore_in_progress boolean = false;
+    BEGIN
+        _is_restore_in_progress = coalesce((SELECT setting::boolean from pg_catalog.pg_settings where name = 'timescaledb.restoring'), false);
+        IF _is_restore_in_progress THEN
+            -- if a restore is in progress, we want the value from the backup, not this hardcoded init value
+            RAISE NOTICE 'restore in progress. skipping insert into _prom_catalog.global_epoch';
+            RETURN;
+        END IF;
+        -- this ensures that pristine and migrated DBs have the same values
+        -- it's also important that current_epoch > delete_epoch
+        INSERT INTO _prom_catalog.global_epoch (current_epoch, delete_epoch, is_unique)
+            VALUES ('2020-01-01 00:00:00 UTC', '1970-01-01 00:00:00 UTC', true);
+    END;
+$block$;
+
+ALTER TABLE _prom_catalog.series ADD COLUMN
+    mark_for_deletion_epoch TIMESTAMPTZ NULL DEFAULT NULL;
```

Comment:
I haven't seen `DEFAULT NULL` used. In some older versions of postgres, specifying a default value on an added column will cause the table to be rewritten. I wasn't sure if this was the case on v12 or if it would happen with `DEFAULT NULL`. So, I tried it...

```
create table foo
( bar int 
, baz text
);
insert into foo (bar, baz) select x, x::text from generate_series(1, 1000) x;
create table b4 as select ctid as _ctid, bar from foo;
alter table foo add column buz timestamptz null default null;
select
  foo.ctid,
  foo.bar,
  b4._ctid,
  b4.bar
from foo
full outer join b4 on (foo.ctid = b4._ctid)
;
```
It does NOT cause the rows to be rewritten. (which is good)

Reply:
Thank you for the analysis!